tests/perception/high_level_semantics_situation_to_developmental_primitive_perception_test.py

```diff
@@ -248,6 +249,56 @@ def test_person_put_ball_on_table():
     )
 
 
+def test_liquid_in_and_out_of_container():
+    juice = SituationObject(ontology_node=JUICE)
+    box = SituationObject(ontology_node=BOX)
+    table = SituationObject(ontology_node=TABLE)
+    two_d_situation = HighLevelSemanticsSituation(
+        ontology=GAILA_PHASE_1_ONTOLOGY,
+        objects=[juice, table],
+        always_relations=[on(juice, table)],
+    )
+
+    two_d_perception = _PERCEPTION_GENERATOR.generate_perception(
+        two_d_situation, chooser=RandomChooser.for_seed(0)
+    )
+
+    three_d_situation = HighLevelSemanticsSituation(
+        ontology=GAILA_PHASE_1_ONTOLOGY,
+        objects=[juice, box],
+        always_relations=[
+            Relation(
+                IN_REGION,
+                juice,
+                Region(
+                    box,
+                    distance=INTERIOR,
+                    direction=Direction(
+                        positive=True, relative_to_axis=GRAVITATIONAL_AXIS
+                    ),
+                ),
+            )
+        ],
+    )
+
+    three_d_perception = _PERCEPTION_GENERATOR.generate_perception(
+        three_d_situation, chooser=RandomChooser.for_seed(0)
+    )
+
+    two_d_frame_property_strings = {
+        f"{p.perceived_object}, {p.binary_property}"
```

Comment:
Please check against the structured objects directly, rather than their stringified versions.